tests/test_application/test_application_utils.py

```diff
@@ -0,0 +1,55 @@
+import collections
+
+import pytest
+from app.notification.application.application_utils import remove_html_tags
+from app.notification.application.application_utils import (
+    sort_questions_and_answers,
+)
+from examplar_data.application_data import test_data_sort_questions_answers
+
+
+@pytest.mark.parametrize(
+    "input_value, expected_response",
+    [
+        ("<p> Hello world </p>", "Hello world"),
+        ("<ul><li>Item 1</li><li>Item 2</li></ul>", "- Item 1\n     - Item 2"),
+        (
+            "<ol><li>Item 1</li><li>Item 2</li></ol>",
+            "1. Item 1\n     2. Item 2",
+        ),
+        ("text without html tags", "text without html tags"),
+        (None, None),
+        (["one", "two"], ["one", "two"]),
+        (True, True),
+        ("https://my-website.com", "https://my-website.com"),
+    ],
+)
+def test_remove_html_tags(app_context, input_value, expected_response):
+
+    response = remove_html_tags(input_value)
+    assert response == expected_response
+
+
+def test_sort_questions_and_answers(app_context):
+    forms = test_data_sort_questions_answers["forms"]
+    form_names = test_data_sort_questions_answers["form_names"]
+
+    response = sort_questions_and_answers(
+        forms, form_names, questions_answers=collections.defaultdict(dict)
+    )
+
+    assert response == test_data_sort_questions_answers["questions_answers"]
+
+
+def test_sort_questions_and_answers_fail(app_context):
+    forms = test_data_sort_questions_answers["incorrect_form_data"]
+    form_names = test_data_sort_questions_answers["form_names"]
+
+    with pytest.raises(Exception) as exc:
+        sort_questions_and_answers(
+            forms, form_names, questions_answers=collections.defaultdict(dict)
+        )
+
+    assert (
+        str(exc.value) == test_data_sort_questions_answers["exception_message"]
+    )
```

Comment:
Hmm was gonna say not a fan of `pytest.raises(Exception)` with the generic `Exception` - but you're asserting on the value afterwards so I think that's a valid solution!

Reply:
Yes, thank you!